recipes/approvaltests.cpp/all/conanfile.py

```diff
@@ -83,6 +83,8 @@ def _std_puttime_required(self):
         return Version(self.version) >= "10.2.0"
 
     def _validate_compiler_settings(self):
+        if self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration("see #7146")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("Currently unsupported by this recipe, see https://github.com/conan-io/conan-center-index/pull/7213#issuecomment-916983425 for more information")
```

Reply:
Perhaps that asm should be replaced with `abort()`

Reply:
they need to merge this fix: https://github.com/catchorg/Catch2/commit/a25c1a24af8bffd35727a888a307ff0280cf9387#diff-ac110b2b11e08efa102d4b37f6a479fb006cdfb9822af039ee07021b349e95bb
btw, it handles too few architectures right now (armv8 and x86/x86_64). maybe something like:
```
#if defined(_MSC_VER)
    DebugBreak();
#elif defined(__GNUC__)
    __builtin_trap();
#else
    abort();
#endif
```
to be more portable.

Reply:
Thank you - this helped me see that the problem is because this ApprovalTests.cpp recipe is depending on an old Catch2 version - I think the latest Catch2 v2.x release should work. I've asked what to do about that in a separate comment.